src/camera_display.cpp

```diff
@@ -374,6 +379,50 @@ void CameraPub::updateFrameRate()
 {
 }
 
+void CameraPub::updateDisplayName()
+{
+    std::string name = name_property_->getStdString();
+    if (name.empty())
+    {
+        setStatus(StatusProperty::Warn, "Display name",
+                  "Should not be empty.");
+        return;
+    }
+
+    ros::NodeHandle nh_private_ = ros::NodeHandle("~"+name);
+    trigger_service_.shutdown();
+    trigger_service_ = nh_private_.advertiseService(camera_trigger_name_, &CameraPub::triggerCallback, this);
+
+    /// Check for service name collision
+    if (trigger_service_.getService().empty())
+    {
+        setStatus(StatusProperty::Warn, "Display name",
+                  "Could not create trigger. Make sure that display name is unique!");
+        return;
+    }
+
+    setStatus(StatusProperty::Ok, "Display name", "OK");
+}
+
+std::string CameraPub::findUnusedDisplayName()
```

Comment:
I'd be curious if there is any existing code for this- `ros::init_options::AnonymousName` just generates a random number (haven't tried it though), roslaunch $(anon foo) creates a name like `foo_hostname_15435_769474121848175325-4`